.github/workflows/verify.yml

```diff
@@ -103,9 +104,27 @@ jobs:
         run: |
           set -euo pipefail
           # shellcheck disable=SC2046
-          go test -json -v $(go list ./... | grep -v '/integration_tests') 2>&1 | tee /tmp/gotest.log | gotestfmt
+          go test -json -v $(go list ./... | grep -v '/integration_tests') -coverprofile=./cover.out -covermode=set -coverpkg=./... 2>&1 | tee /tmp/gotest.log | gotestfmt
         env:
           GH_TOKEN: ${{ secrets.GH_TOKEN }}
+      - name: Report test coverage
+        uses: vladopajic/go-test-coverage@v2
+        continue-on-error: true
+        with:
+          config: ./.github/.testcoverage.yml
+      - name: Generate html test coverage
+        shell: bash
+        run: go tool cover -html=cover.out -o coverage.html
+      - name: Upload html coverage
+        uses: actions/upload-artifact@v4
+        with:
+          name: unit_coverage
+          path: coverage.html
```

Comment:
Do we need .html reports for unit and integration tests separately, or we can just leave one file with merged results?